readthedocs/vcs_support/backends/git.py

```diff
@@ -53,38 +55,55 @@ def set_remote_url(self, url):
 
     def update(self):
         # Use checkout() to update repo
+        # TODO: See where we call this
         self.checkout()
 
     def repo_exists(self):
         code, _, _ = self.run('git', 'status', record=False)
         return code == 0
 
-    def are_submodules_available(self):
-        """
-        Test whether git submodule checkout step should be performed.
-
-        .. note::
-
-            Temporarily, we support skipping these steps as submodule step can
-            fail if using private submodules. This will eventually be
-            configureable with our YAML config.
-        """
-        # TODO remove with https://github.com/rtfd/readthedocs-build/issues/30
+    def are_submodules_available(self, config):
+        """Test whether git submodule checkout step should be performed."""
+        # TODO remove this after users migrate to a config file
         from readthedocs.projects.models import Feature
-        if self.project.has_feature(Feature.SKIP_SUBMODULES):
+        submodules_in_config = (
+            config.submodules.exclude != ALL or
+            config.submodules.include
+        )
+        if (self.project.has_feature(Feature.SKIP_SUBMODULES) or
+                not submodules_in_config):
             return False
+
+        # Keep compatibility with previous projects
         code, out, _ = self.run('git', 'submodule', 'status', record=False)
```

Comment:
We were running this code to check for submodules previously, so we need to keep the same behavior here